.github/workflows/ci.yaml

```diff
@@ -36,19 +36,18 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        # 3.9 is the current >= 1.3.0 python version
         python-version: ["3.9"]
     steps:
     - uses: actions/checkout@v3
-    - name: Set up python ${{ matrix.python-version }}
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
         cache: pip
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip~=23.2.1
-        pip install -r dev-requirements.txt
+        pip install pre-commit==3.6.0
```

Comment:
Shouldn't this match the requirement in `dev-requirements.txt`?
Either `pre-commit~=3.6.0` or `pre-commit~=3.6` would probably be in order.

Reply:
I wonder whether the linter really doesn't need the requirements installed to properly validate (e.g. to validate correct number of arguments is passed to a function imported from a module in required library).

Reply:
> Shouldn't this match the requirement in `dev-requirements.txt`? Either `pre-commit~=3.6.0` or `pre-commit~=3.6` would probably be in order.

I preferred stability in the CI and therefore specified `==`. I can change it to `~=`, but then it's less predictable. Let me know what you think.



> I wonder whether the linter really doesn't need the requirements installed to properly validate (e.g. to validate correct number of arguments is passed to a function imported from a module in required library).

"the linter" - what do you mean by that? pre-commit? Ruff?
The hooks are not local, and pre-commit manages the environments for each hook separately.
Please clarify this point with an explicit example.

Reply:
> "the linter" - what do you mean by that? pre-commit? Ruff?

Ruff.

> The hooks are not local, and pre-commit manages the environments for each hook separately.

Not sure what this means or how it applies here.

> Please clarify this point with an explicit example.

I did, in parentheses. Doesn't ruff need to have the dependencies installed to know if a function is called with valid arguments?

Reply:
Ok, never mind, it just doesn't take the arguments passed at all...

Reply:
> Ok, never mind, it just doesn't take the arguments passed at all...

Right. There are other tools that check that, for example: mypy ([ML-5304](https://jira.iguazeng.com/browse/ML-5304)). Pylint also does some static analysis checks, but that part is not in Ruff AFAIK.